api/src/main/java/org/openmrs/module/appointments/notification/impl/DefaultSmsSender.java

```diff
@@ -0,0 +1,180 @@
+            String sender = "&" + smsSessionProperties.getProperty("sms.key.sender") + "=" + URLEncoder.encode(smsSessionProperties.getProperty("sms.sender"), "UTF-8");
+            String number = "&" + smsSessionProperties.getProperty("sms.key.phonenumber") + "=" + URLEncoder.encode(recipientMobileNumber, "UTF-8");
+
+            if (StringUtils.isNotBlank(smsSessionProperties.getProperty("sms.key.password"))) {
+                password = "&" + smsSessionProperties.getProperty("sms.key.password") + "=" + URLEncoder.encode(smsSessionProperties.getProperty("sms.password") , "UTF-8");
+            }
+            if (StringUtils.isNotBlank(smsSessionProperties.getProperty("sms.key.entity.identifier"))) {
+                entityId = "&" + smsSessionProperties.getProperty("sms.key.entity.identifier") + "=" + URLEncoder.encode(smsSessionProperties.getProperty("sms.entity.identifier"), "UTF-8");
+            }
+            if (StringUtils.isNotBlank(smsSessionProperties.getProperty("sms.key.template.identifier")) && "Virtual".equalsIgnoreCase(smsType)) {
+                templateId = "&" + smsSessionProperties.getProperty("sms.key.template.identifier") + "=" + 
+                    URLEncoder.encode(smsSessionProperties.getProperty("sms.appointment.teleconsultation.template.identifier"), "UTF-8");
+                // Dear %s,\nTele-Consultation booked for %s %s\nVideo link: %s Bahmni Hospital(IPLit)
+                msgText = String.format(smsSessionProperties.getProperty("sms.appointment.teleconsultation.message.template"), patientName,
+                    fmtDateTele.format(startDateTime), fmtTime.format(startDateTime), teleLink);
+            }
+            if (StringUtils.isNotBlank(smsSessionProperties.getProperty("sms.key.template.identifier")) && "WalkIn".equalsIgnoreCase(smsType)) {
+                templateId = "&" + smsSessionProperties.getProperty("sms.key.template.identifier") + "=" + 
+                    URLEncoder.encode(smsSessionProperties.getProperty("sms.appointment.walkin.template.identifier"), "UTF-8");
+                // Dear %s,\n Your appointment is booked for %s on %s at %s by Bahmni Hospital (powered by IPLit)
+                msgText = String.format(smsSessionProperties.getProperty("sms.appointment.walkin.message.template"), patientName,
+                    serviceName, fmtDateAppt.format(startDateTime), fmtTime.format(startDateTime));
+            }
+            if (StringUtils.isNotBlank(smsSessionProperties.getProperty("sms.params.addons"))) {
+                addOnParams = "&" + smsSessionProperties.getProperty("sms.params.addons");
+            }
+            String message = "&" + smsSessionProperties.getProperty("sms.key.message") + "=" + URLEncoder.encode(msgText, "UTF-8");
+            requestUrl += apiKey + message + sender + number + password + entityId + templateId + addOnParams;
```

Comment:
sending the msg itself through the URL  is not a standard way of passing the sms. Can you check if you can pass through the body/payload .. 
Also, its probably better to use a closeable HTTP Client, and you can use try with resources block